tests/org/intellij/erlang/compilation/ErlangRebuildInDifferentModuleTest.java

```diff
@@ -0,0 +1,73 @@
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.roots.ModuleRootModificationUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+
+import static org.intellij.erlang.compilation.ErlangModuleTextGenerator.behaviour;
+import static org.intellij.erlang.compilation.ErlangModuleTextGenerator.module;
+import static org.intellij.erlang.compilation.ErlangModuleTextGenerator.pt;
+
+public class ErlangRebuildInDifferentModuleTest extends ErlangCompilationTestBase {
+  public void testRebuildWithParseTransformInDifferentModule() throws Exception {
+    Module otherModule = createModuleInOwnDirectoryWithSourceAndTestRoot("other");
+    ModuleRootModificationUtil.addDependency(myModule, otherModule);
+    VirtualFile parseTransform = addSourceFile(otherModule, "parse_transform1.erl", pt("parse_transform1").build());
+    VirtualFile sourceFile = addSourceFile(myModule, "module1.erl", module("module1").pt("parse_transform1").build());
+    doTestCrossModulesDependency(otherModule, parseTransform, sourceFile);
+  }
+
+  public void testRebuildWithBehaviourInDifferentModule() throws Exception {
+    Module otherModule = createModuleInOwnDirectoryWithSourceAndTestRoot("other");
+    ModuleRootModificationUtil.addDependency(myModule, otherModule);
+    ErlangModuleTextGenerator.BehaviourBuilder behaviour = behaviour("behaviour1").callback("foo", 0);
+    VirtualFile behaviourFile = addSourceFile(otherModule, "behaviour1.erl", behaviour.build());
+    VirtualFile sourceFile = addSourceFile(myModule, "module1.erl", module("module1").behaviour(behaviour).build());
+    doTestCrossModulesDependency(otherModule, behaviourFile, sourceFile);
+  }
+
+  public void testRebuildWithIncludesInDifferentModule() throws Exception {
+    Module otherModule = createModuleInOwnDirectoryWithSourceAndTestRoot("other");
+    ModuleRootModificationUtil.addDependency(myModule, otherModule);
+    VirtualFile headerFile = addSourceFile(myModule, "header.hrl", "");
+    VirtualFile sourceFile = addSourceFile(myModule, "module1.erl", module("module1").include("header.hrl").build());
+    doTestCrossModulesDependency(otherModule, headerFile, sourceFile);
+  }
+
+  public void doTestCrossModulesDependency(Module otherModule,
+                                           VirtualFile dependency,
+                                           VirtualFile sourceFile) throws Exception {
+    CompilationRunner compilationRunner = new CompilationRunner(myModule, otherModule);
+    try {
+      compilationRunner.compile();
+      assertSourcesCompiled(myModule, false);
+      assertSourcesCompiled(otherModule, false);
+      long sourceModificationTime = lastOutputModificationTime(myModule, sourceFile);
+      compilationRunner.touch(dependency);
+      compilationRunner.compile();
+      assertSourcesCompiled(myModule, false);
+      assertSourcesCompiled(otherModule, false);
+      assertNotSame(sourceModificationTime, lastOutputModificationTime(myModule, sourceFile));
```

Comment:
This is not the assertion you wanted to use. It will do reference comparison, whereas you want to check for inequality.